vms/platformvm/state.go

```diff
@@ -389,20 +389,22 @@ func (vm *VM) removeUTXO(db database.Database, utxoID ids.ID) error {
 // Only returns UTXOs after [start].
 // Returns at most [limit] UTXO IDs.
 // Returns nil if no UTXOs reference [addr].
-func (vm *VM) getReferencingUTXOs(db database.Database, addr []byte, start ids.ID, limit int) (ids.Set, error) {
-	toFetch := limit
-	utxoIDs := ids.Set{}
+func (vm *VM) getReferencingUTXOs(db database.Database, addr []byte, start ids.ID, limit int) ([]ids.ID, error) {
+
+	idSlice := []ids.ID(nil)
+
 	iter := prefixdb.NewNested(addr, db).NewIteratorWithStart(start.Bytes())
 	defer iter.Release()
-	for toFetch > 0 && iter.Next() {
-		if utxoID, err := ids.ToID(iter.Key()); err != nil {
+	numFetched := 0
+	for numFetched < limit && iter.Next() {
+		if keyID, err := ids.ToID(iter.Key()); err != nil {
 			return nil, err
-		} else if !utxoID.Equals(start) {
-			utxoIDs.Add(utxoID)
-			toFetch--
+		} else if !bytes.Equal(keyID.Bytes(), start.Bytes()) {
```

Comment:
This can be `keyID != start`

Reply:
Ended up reverting it as I can't compare those types 🤔  
It was creating an infinite loop on https://github.com/ava-labs/avalanchego/blob/b39d1fa003f1511190d1b031cce966cb46cd1161/vms/platformvm/state.go#L402

I could do `keyID.ID` but just went with this as it's the same as we have on the avm .

Reply:
I think this PR currently doesn't have the most update to code, and Dan was assuming it did. I think his comment should be implemented once this PR has merged in the existing changes on `dev`.

Reply:
Yep that was it 👍